libtextsecure/message_receiver.js

```diff
@@ -1129,6 +1129,17 @@ MessageReceiver.prototype.extend({
       window.storage.remove('secondaryDeviceStatus');
       window.storage.put('isSecondaryDevice', true);
       Whisper.events.trigger('secondaryDeviceRegistration');
+      // Update profile name
+      if (dataMessage) {
```

Comment:
```suggestion
      if (dataMessage && dataMessage.profile) {
```
This would simplify the branching conditions and you can delete the nested if

Reply:
Nice simplification :)